src/msvc_helper-win32.cc

```diff
@@ -58,14 +58,19 @@ int CLWrapper::Run(const string& command, string* output) {
     Win32Fatal("SetHandleInformation");
 
   PROCESS_INFORMATION process_info = {};
-  STARTUPINFOA startup_info = {};
+  STARTUPINFO startup_info = {};
   startup_info.cb = sizeof(STARTUPINFOA);
   startup_info.hStdInput = nul;
   startup_info.hStdError = ::GetStdHandle(STD_ERROR_HANDLE);
   startup_info.hStdOutput = stdout_write;
   startup_info.dwFlags |= STARTF_USESTDHANDLES;
 
-  if (!CreateProcessA(NULL, (char*)command.c_str(), NULL, NULL,
+#ifdef _WIN32
+  wstring commands = Utf8ToWide(command);
+  if (!CreateProcessW(NULL, (wchar_t*)commands.c_str(), NULL, NULL,
```

Comment:
Big no no.

Cannot assume that the c_str() returned by this call can be safely treated as non-const.

Given @jhasse 's statement that C++11 inside of #ifdef _WIN32 is acceptable, use commands.data() instead. C++11 strings are always nul terminated, and calling commands.data() would give you a mutable, nul terminated wchar_t string to pass to CreateProcess.

Reply:
data() also returns const wchar* though.

Reply:
*sigh*. You're right.

***C++17*** is where the .data() function returned a non-const.

https://en.cppreference.com/w/cpp/string/basic_string/data
https://stackoverflow.com/questions/1919626/can-i-get-a-non-const-c-string-back-from-a-c-string

We're so far behind, it's difficult to keep track of the specific version where these changes to the standard were made.

--------------------------------------

So here we have to do one of two things.

1. Continue with the assumption of C++11 for Windows builds, and therefore use the knowledge that the internal storage of strings in C++11 are contiguous, and nul terminated, to access the string data.
2. Assume C++03, and copy the string to a mutable buffer (with length checking to prevent a buffer over run).

To do 1), use `&command[0]`